build.sbt

```diff
@@ -1,24 +1,28 @@
 organization in ThisBuild := "com.example"
-version in ThisBuild := "1.0-SNAPSHOT"
 
 // the Scala version that will be used for cross-compiled libraries
 scalaVersion in ThisBuild := "2.12.8"
 
+// Update the version generated by sbt-dynver to remove any + characters, since these are illegal in docker tags
+version in ThisBuild ~= (_.replace('+', '-'))
+dynver in ThisBuild ~= (_.replace('+', '-'))
+
 lazy val `shopping-cart-java` = (project in file("."))
-  .aggregate(`shopping-cart-api`, `shopping-cart-impl`, `inventory-api`, `inventory-impl`)
+  .aggregate(`shopping-cart-api`, `shopping-cart`, `inventory-api`, inventory)
 
 lazy val `shopping-cart-api` = (project in file("shopping-cart-api"))
-  .settings(common: _*)
+  .settings(common)
   .settings(
     libraryDependencies ++= Seq(
       lagomJavadslApi,
       lombok
     )
   )
 
-lazy val `shopping-cart-impl` = (project in file("shopping-cart-impl"))
+lazy val `shopping-cart` = (project in file("shopping-cart"))
   .enablePlugins(LagomJava)
-  .settings(common: _*)
+  .settings(common)
+  .settings(dockerSettings)
```

Comment:
JFYI .settings takes SettingsDefinition*, of which both Setting and Seq[Setting] are a subtype of (the former via inheritance, the latter via an implicit scope conversion).

So, TL;DR, you can mix and match settings with seq of settings without having to open and close multiple .settings.